src/Views/BucketPanel/EntryDetail.tsx

```diff
@@ -181,6 +285,153 @@ export default function EntryDetail(props: Readonly<Props>) {
           onRemoved={() => history.push(`/buckets/${bucketName}`)}
         />
       )}
+
+      <Modal
+        title="Upload File"
+        open={isUploadModalVisible}
+        onCancel={() => {
+          setIsUploadModalVisible(false);
+          setUploadError("");
+          uploadForm.resetFields();
+          setUploadFile(null);
+          setLabels([]);
+        }}
+        footer={null}
+        width={600}
+      >
+        <Form
+          form={uploadForm}
+          onFinish={handleUpload}
+          layout="vertical"
+          style={{ padding: "20px 0" }}
+        >
+          <Form.Item style={{ marginBottom: "24px" }}>
+            <Upload.Dragger
+              beforeUpload={(file) => {
+                // Add file size check (e.g., 10MB limit)
+                const maxSize = 10 * 1024 * 1024; // 10MB
```

Comment:
it's to little, let's do 1GB.

We use the SI: 10 * 1000 * 1000 MB, what you count is 10MiB.